src-self-hosted/translate_c.zig

```diff
@@ -879,205 +875,166 @@ fn transBinaryOperator(
                 return maybeSuppressResult(rp, scope, result_used, &div_trunc_node.base);
-            } else {
-                // unsigned/float division uses the operator
-                const node = try transCreateNodeInfixOp(rp, scope, stmt, .Div, .Slash, "/", true);
-                return maybeSuppressResult(rp, scope, result_used, node);
             }
         },
         .Rem => {
             if (!cIsUnsignedInteger(qt)) {
                 // signed integer division uses @rem
                 const rem_node = try transCreateNodeBuiltinFnCall(rp.c, "@rem");
-                const lhs = try transExpr(rp, scope, ZigClangBinaryOperator_getLHS(stmt), .used, .l_value);
-                try rem_node.params.push(lhs);
+                try rem_node.params.push(try transExpr(rp, scope, ZigClangBinaryOperator_getLHS(stmt), .used, .l_value));
                 _ = try appendToken(rp.c, .Comma, ",");
                 const rhs = try transExpr(rp, scope, ZigClangBinaryOperator_getRHS(stmt), .used, .r_value);
                 try rem_node.params.push(rhs);
                 rem_node.rparen_token = try appendToken(rp.c, .RParen, ")");
                 return maybeSuppressResult(rp, scope, result_used, &rem_node.base);
-            } else {
-                // unsigned/float division uses the operator
-                const node = try transCreateNodeInfixOp(rp, scope, stmt, .Mod, .Percent, "%", true);
-                return maybeSuppressResult(rp, scope, result_used, node);
             }
         },
         .Shl => {
             const node = try transCreateNodeShiftOp(rp, scope, stmt, .BitShiftLeft, .AngleBracketAngleBracketLeft, "<<");
-            return maybeSuppressResult(rp, scope, result_used, TransResult{
-                .node = node,
-                .child_scope = scope,
-                .node_scope = scope,
-            });
+            return maybeSuppressResult(rp, scope, result_used, node);
         },
         .Shr => {
             const node = try transCreateNodeShiftOp(rp, scope, stmt, .BitShiftRight, .AngleBracketAngleBracketRight, ">>");
-            return maybeSuppressResult(rp, scope, result_used, TransResult{
-                .node = node,
-                .child_scope = scope,
-                .node_scope = scope,
-            });
+            return maybeSuppressResult(rp, scope, result_used, node);
+        },
+        .LAnd => {
+            const node = try transCreateNodeBoolInfixOp(rp, scope, stmt, .BoolAnd, result_used, true);
+            return maybeSuppressResult(rp, scope, result_used, node);
+        },
+        .LOr => {
+            const node = try transCreateNodeBoolInfixOp(rp, scope, stmt, .BoolOr, result_used, true);
+            return maybeSuppressResult(rp, scope, result_used, node);
+        },
+        else => {},
+    }
+    const lhs_node = try transExpr(rp, scope, ZigClangBinaryOperator_getLHS(stmt), .used, .l_value);
+    switch (op) {
+        .PtrMemD, .PtrMemI, .Cmp => return revertAndWarn(
```

Comment:
I think `PtrMemD` and `PtrMemI` should be unreachable, because they're C++ things?
https://github.com/llvm-mirror/clang/blob/master/include/clang/AST/OperationKinds.def#L357

Also the spaceship `<=>` (`Cmp`) operator:
https://github.com/llvm-mirror/clang/blob/master/include/clang/AST/OperationKinds.def#L370

Reply:
Definitely agree, thanks for looking them up.